lib/samples/generate_offline_map/generate_offline_map_sample.dart

```diff
@@ -0,0 +1,227 @@
+//
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'dart:io';
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+
+import '../../utils/sample_state_support.dart';
+
+class GenerateOfflineMapSample extends StatefulWidget {
+  const GenerateOfflineMapSample({super.key});
+
+  @override
+  State<GenerateOfflineMapSample> createState() =>
+      _GenerateOfflineMapSampleState();
+}
+
+class _GenerateOfflineMapSampleState extends State<GenerateOfflineMapSample>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // Create a Graphic to show the area to be taken offline.
+  final _regionGraphic = Graphic();
```

Comment:
I think we need to use a UI component for this. When you rotate the map view right now, the graphic red box rotates around and changes size whereas I think it should be fixed. Other platforms have used `Rectangle` API equivalents to fix it in place. There is a `Rect` class in Dart but maybe `SizedBox` or `Positioned` widgets would work. Before we change it though I've posed the question to the other SDKs to confirm, and will also update the design if this is the case to make it less vague.

Reply:
The Kotlin version uses a Graphic and viewpointChanged the same way: https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/blob/74a3c10646b5f5e39644c995fc1e28b3cb9515f0/generate-offline-map/src/main/java/com/esri/arcgismaps/sample/generateofflinemap/MainActivity.kt#L71C17-L71C29

I tried using SizedBox/Positioned but it gets really hairy trying to map a Widget's on-screen position to the map view. It might be possible, but you end up with a bunch of code like our internal `ViewMetrics.fromContext`. It would be a big detour from the tutorial flow.

Reply:
Yeah the Kotlin sample has the same behavior issues as well though. To me it's not that easy to use right now because as soon as you start interacting with the map view you lose your rectangle. Maybe we can disable map view rotation, might improve the behavior if it's not a simple thing to achieve with a widget in Flutter? Agree we want to get the balance in the sample.

Reply:
I re-worked it with the fixed Widget outline and the screen-to-map conversion logic.